tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,494 @@
+
+from tests.common import MockDependency
+
+CLIENT_ENTITY_ID = 'media_player.client_dvr'
+MAIN_ENTITY_ID = 'media_player.main_dvr'
+IP_ADDRESS = '127.0.0.1'
+
+DISCOVERY_INFO = {
+    'host': IP_ADDRESS,
+    'serial': 1234
+}
+
+LIVE = {
+    "callsign": "CNNHD",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": False,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield"
+}
+
+LOCATIONS = [
+    {
+        'locationName': 'Main DVR',
+        'clientAddr': DEFAULT_DEVICE
+    }
+]
+
+RECORDING = {
+    "callsign": "CNNHD",
+    "date": "20181110",
+    "duration": 3600,
+    "isOffAir": False,
+    "isPclocked": 1,
+    "isPpv": False,
+    "isRecording": True,
+    "isVod": False,
+    "major": 202,
+    "minor": 65535,
+    "offset": 1,
+    "programId": "102454523",
+    "rating": "No Rating",
+    "startTime": 1541876400,
+    "stationId": 3900947,
+    "title": "CNN Newsroom With Fredricka Whitfield",
+    'uniqueId': '12345',
+    'episodeTitle': 'CNN Recorded'
+}
+
+WORKING_CONFIG = {
+    'media_player': {
+        'platform': 'directv',
+        CONF_HOST: IP_ADDRESS,
+        CONF_NAME: 'Main DVR',
+        CONF_PORT: DEFAULT_PORT,
+        CONF_DEVICE: DEFAULT_DEVICE
+    }
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def platforms():
+    """Fixture for setting up test platforms."""
+    config = {
+        'media_player': [{
+            'platform': 'directv',
+            'name': 'Main DVR',
+            'host': IP_ADDRESS,
+            'port': DEFAULT_PORT,
+            'device': DEFAULT_DEVICE
+        }, {
+            'platform': 'directv',
+            'name': 'Client DVR',
+            'host': IP_ADDRESS,
+            'port': DEFAULT_PORT,
+            'device': '1'
+        }]
+    }
+
+    async def setup_test_platforms(hass):
+        with MockDependency('DirectPy'), \
+             patch('DirectPy.DIRECTV', new=mockDIRECTVClass):
+            await async_setup_component(hass, mp.DOMAIN, config)
+            await hass.async_block_till_done()
+
+        main_media_entity = hass.data['media_player'].get_entity(
+            MAIN_ENTITY_ID)
+
+        client_media_entity = hass.data['media_player'].get_entity(
+            CLIENT_ENTITY_ID)
+
+        # Set the client so it seems a recording is being watched.
+        client_media_entity.dtv.attributes = RECORDING
+        # Clients do not support turning on, setting it as client is on here.
+        client_media_entity.dtv._standby = False
+
+        main_media_entity.schedule_update_ha_state(True)
+        client_media_entity.schedule_update_ha_state(True)
+        await hass.async_block_till_done()
+
+        return (main_media_entity, client_media_entity)
```

Comment:
Don't return the entities. We should use the states from the state machine to assert state in the tests.